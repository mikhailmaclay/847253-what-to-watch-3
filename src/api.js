// Libraries
import axios from 'axios';
// Constants and utils
import {Config} from './constants/consts';

const Error = {
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  SERVER_UNAVAILABLE: 503
};

export const createAPI = (onNoResponse = () => {}, onUnauthorized = () => {}, onBadRequest = () => {}, onServerUnavailable = () => {}) => {
  const api = axios.create({
    baseURL: `https://htmlacademy-react-3.appspot.com/wtw`,
    timeout: Config.REQUEST_TIMEOUT,
    withCredentials: true
  });

  const onLoad = (response) => response;

  const onError = (error) => {
    const {response} = error;

    if (!response) {
      onNoResponse();

      throw error;
    }

    const {status} = response;

    switch (status) {
      case Error.UNAUTHORIZED:
        onUnauthorized();

        throw error;

      case Error.BAD_REQUEST:
        onBadRequest();

        throw error;

      case Error.SERVER_UNAVAILABLE:
        onServerUnavailable();

        throw error;

      default:
        throw error;
    }
  };

  api.interceptors.response.use(onLoad, onError);

  return api;
};
