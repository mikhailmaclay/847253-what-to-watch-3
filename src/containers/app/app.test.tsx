// Libraries
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
// Components
import AppContainer from './app';
//
import {createAPI} from '../../api';

const api = createAPI();
const mockStore = configureStore([thunk.withExtraArgument(api)]);

describe(`<AppContainer/>`, () => {
  const store = mockStore({});

  it(`Should render correctly`, () => {
    const result = renderer.create(
        <Provider store={store}>
          <AppContainer/>
        </Provider>
    ).toJSON();

    expect(result).toMatchSnapshot();
  });
});
