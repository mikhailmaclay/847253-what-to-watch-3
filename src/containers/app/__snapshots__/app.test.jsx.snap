// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppContainer/> Should render correctly 1`] = `null`;
